src/k8s-engine/__tests__/k3sHelper.spec.ts

```diff
@@ -191,4 +194,30 @@ describe(K3sHelper, () => {
     expect(() => subject.fullVersion('1.2.4')).toThrow('1.2.4');
     expect(() => subject.fullVersion('invalid version')).toThrow('not a valid version');
   });
+
+  describe('initialize', () => {
+    it('should finish initialize without network if cache is available', async() => {
+      const writer = new K3sHelper();
+
+      writer['versions'] = { 'v1.0.0': new semver.SemVer('v1.0.0') };
+      await writer['writeCache']();
```

Comment:
Why is this not `await writer.writeCache();` ?

Reply:
Because `writeCache` is `protected`; TypeScript allows this notation to get around access restrictions (I guess they're more recommendations).  Note that the TypeScript compiler actually still knows you're accessing it (so if it's minimizing it'll keep things working).